scripts/lint_jjb.py

```diff
@@ -155,6 +164,113 @@ def check_retention(job, in_file):
     return return_value
 
 
+# Ensure all jobs have a valid timed triggers entry
+# Structure:
+#   job:
+#     triggers:
+#      - timed: "@daily"
+def check_timed_trigger_value(trigger):
+
+    # We cannot parse macros, so we assume that
+    # they are ok.
+    if '{' in trigger and '}' in trigger:
+        return_value = 0
+    else:
+        # Jenkins supports the use of hashes, denoted by the symbol H,
+        # in the trigger to allow the load to be spread on the system.
+        # The crontab library does not know how to validate them, so
+        # we replace them with 0.
+        _trigger = trigger.replace('H', '0')
+
+        if crontab.CronSlices.is_valid(_trigger):
```

Comment:
`crontab.CronSlices.is_valid` appears to return `True` for most objects that aren't a string as well as for an empty string:

```
>>> crontab.CronSlices.is_valid("")
True
>>> crontab.CronSlices.is_valid(None)
True
>>> crontab.CronSlices.is_valid(0)
True
>>> crontab.CronSlices.is_valid(False)
True
```

This function assumes the value set will be a string so this mostly shouldn't matter but if it's an empty string, what does that mean? Is the job disabled? Will the job fail?

Reply:
It looks like the python-crontab library is quite bad at validating input. The croniter library is far better, so switching to it.